client/blocks/site/index.jsx

```diff
@@ -145,3 +147,10 @@ export default React.createClass( {
 		);
 	}
 } );
+
+export default connect( ( state, { site } ) => {
+	if ( isNumber( site ) ) { // Accept site IDs
```

Comment:
This feels a bit strange to me, I think adding `siteId` would be clearer here.

So both 
```jsx
<Site site={ site } />
```

and 
```jsx
<Site siteId={ siteId } />
```

would be accepted usages. We can then do:
```jsx
const mapStateToProps = ( state, { siteId, site } ) => {
    const site = siteId ? getSite( state, site ) : site;
    return {
        site
    };
}
```